Proyecto II: desarrollar un script que, a traves de funciones, permita obtener informacion estadıstica descriptiva de una tabla dada.


In [ ]:
!pip install wooldridge
import wooldridge as wd

     |████████████████████████████████| 5.1 MB 5.1 MB/s 


In [ ]:
df = wd.data('wage1')
df = df[['wage','educ']]
df.head(10)
#Tabla dada

wage  educ
0   3.10    11
1   3.24    12
2   3.00    11
3   6.00     8
4   5.30    12
5   8.75    16
6  11.25    18
7   5.00    12
8   3.60    12
9  18.18    17

In [ ]:
#Script
class estadisticos:

  def __init__(self,x):
    self.x=x
  
  def media(self):
    return round(sum(self.x)/len(self.x),3)

  def mediana(self):
    mediana=0
    self.x.sort()
    tamaño=len(self.x)

    if tamaño%2!=0:
      mediana+=self.x[tamaño//2]
      return round(mediana, 3)

    else:
      num_1=self.x[tamaño//2]
      num_2=self.x[tamaño//2-1]
      mediana+=(num_1+num_2)/2
      return round(mediana, 3)

  def desviacion_estandar(self):
    nume=0
    for i in self.x:
      nume += (i - estadisticos.media(self))**2

    denom = nume/(len(self.x)-1)
    return round(denom**0.5 ,3)

  def varianza(self):
    vari = estadisticos.desviacion_estandar(self)**2
    return round(vari,3)
  
  def coef_var(self):
    cv = estadisticos.desviacion_estandar(self)/abs(estadisticos.media(self))
    return round(cv,3)

In [ ]:
wage_ = estadisticos(df['wage'].to_list())
educ_ = estadisticos(df["educ"].to_list())

In [ ]:
#Data Salario
print(wage_.media())
print(wage_.mediana())
print(wage_.desviacion_estandar())
print(wage_.varianza())
print(wage_.coef_var())

5.896
4.65
3.693
13.638
0.626


In [ ]:
#Data educación
print(educ_.media())
print(educ_.mediana())
print(educ_.desviacion_estandar())
print(educ_.varianza())
print(educ_.coef_var())

12.563
12.0
2.769
7.667
0.22


#Conclusiones: 
En promedio las personas tiene un grado de escolaridad cercano al 13 con una cifra representativa de 12.563 y asimismo reciben un salario de 5.896 aproximadamente. No obstante, identificamos que casi la mitad de la población reporta a una escolaridad  de nivel 12 y en tanto al salario la cifra que más se reporta corresponde a un pago de 4.65. Por otra parte, en tanto a la dipersión de los datos podemos identificar que la desviación estandar es mayor para las cifras proporcionadas en los salarios que en los datos de educación, aunque el grado de dispersion entre ambos discierne en 0.924, si se refleja  un cambio representativo en la varianza en la cual, para ambos casos encontramos una gran variabilidad de los datos con respecto a su media. Finalmente,  podemos mencionar que la variable de salario tiene una muestra muy dispersa y su media pierde confiabilidad dado que el dato se aproxima más a 1, caso contrario se evidencia en la variable de educación en la cual existe poca variabilidad de los datos. 
